src/clr-angular/data/tree-view/tree-node.spec.ts

```diff
@@ -163,19 +165,34 @@ export default function(): void {
     describe('Template API', function() {
       spec(ClrTreeNode, TestComponent, ClrTreeViewModule, { imports: [NoopAnimationsModule, ClrIconModule] });
 
-      it(
-        'offers a [(clrSelected)] two-way binding',
-        fakeAsync(function(this: Context) {
-          this.hostComponent.selected = ClrSelectedState.SELECTED;
-          this.detectChanges();
-          expect(this.clarityDirective.selected).toBe(ClrSelectedState.SELECTED);
-          this.clarityDirective.selected = ClrSelectedState.INDETERMINATE;
-          // We need fakeAsync and tick because the EventEmitter is asynchronous
-          tick();
-          // I don't know why Typescript forces me to cast this
-          expect<ClrSelectedState>(this.hostComponent.selected).toBe(ClrSelectedState.INDETERMINATE);
-        })
-      );
+      it('offers a [(clrSelected)] two-way binding', function(this: Context) {
+        // Here we are testing the child two-way binding, which should be updated by two-way always.
+        const child = this.fixture.debugElement.query(By.directive(ClrTreeNode));
+        this.hostComponent.selected = ClrSelectedState.SELECTED;
+        this.detectChanges();
+        expect(this.clarityDirective.selected).toBe(ClrSelectedState.SELECTED);
+        expect(child.componentInstance.selected).toBe(ClrSelectedState.SELECTED);
+        this.clarityDirective.selected = ClrSelectedState.UNSELECTED;
+        this.detectChanges();
+        expect(child.componentInstance.selected).toBe(ClrSelectedState.UNSELECTED);
```

Comment:
Testing the child here seems to have nothing to do with the current unit test. I get that the line 
```typescript
this.clarityDirective.selected = ClrSelectedState.UNSELECTED;
```
doesn't actually trigger the output for the parent, but this test relies on the entire logic of downward propagation of selection for eager trees, which feels wrong for a pure two-way binding test.

I think the next unit test you added covers the two-way binding, so you can remove this one or change it to describe more specifically what you're testing (presumably chocolate errors?).

Reply:
Since the selection will propigate to the child, I'm testing that as the two way binding to get around the input change limitation I'm testing in the other test. If you're happy with the second test then I'll just remove this one.